Fix level dependency installation on a recent Linux + bugfix to overwrite options parameter

ludovic-gasc opened the pull request:
Hi,

Thanks for your JS library, it still works pretty well in 2020 with the latest version of NodeJS and Chrome.

However, to be able to use it, I needed to change two elements in the source code, to be able to install the latest version of level, and fix a bug with options handling.

Could you merge this change and update the package on npmjs.com ?

Thanks

ludovic-gasc committed:
Fix attempt during installation problem

ludovic-gasc committed:
Invert parameters for options object fusion

Without this, it's impossible to override DEFAULT_OPTIONS values

ludovic-gasc committed:
Increase version number

ludovic-gasc committed:
Remove broken image in README.md

carlosascari commented on index.js:
```diff
@@ -107,7 +107,7 @@ class ChromeCacheFile {
 class ChromeCacheReader {
   constructor(options={}) {
-    this.options = Object.assign(options, DEFAULT_OPTIONS)
+    this.options = Object.assign(DEFAULT_OPTIONS, options)
```
Here you made the `DEFAULT_OPTIONS` object the target to be modified by the `options` object

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

As I understand it, this will change the `DEFAULT_OPTIONS`, which should be a constant/frozen object.

~~Perhaps there is something I don't understand, can you please explain to me why merging the `options` object **into** the `DEFAULT_OPTIONS` object a bugfix?~~

## I see the issue, the fix should be:

```
Object.assign({}, DEFAULT_OPTIONS, options)
```

ludovic-gasc replied:
Hi @carlosascari,

Indeed, it's a good practice to use an empty object at the beginning of the chain.

FYI, the first fix I quickly made was with the spread operator ... => https://medium.com/javascript-in-plain-english/8-ways-to-use-spread-operator-in-javascript-b66fcf016efe
But this operator is present only with ES6+.

Thanks for the merge and to have published a new release on npmjs.org

Have a nice week-end :-)

carlosascari reviewed (changes requested):
Hello,

~~I'd be happy to update the package, however I need to understand the options bugfix you are proposing, here:~~

https://github.com/carlosascari/chrome-cache-reader/pull/1/commits/ceb2358e878296a03d33aaef7851f3f2ab6c53d5

~~I left a comment with my doubts. Essentially the fix you are proposing modifies a constant object, the defaults object, which should not be modified, since for example, a second instance of the ChromeCacheReader class would use a modified defaults object.~~

The package has been updated on npmjs

Thank you :+1:

carlosascari committed:
Update index.js

Fixed options Object.assignment so as to properly create a new object with the user set options and default options merged, without modifying the original default options object
